crates/notary/server/src/settings.rs

```diff
@@ -0,0 +1,39 @@
+use config::{Config, ConfigError, Environment, File};
+use std::path::Path;
+use crate::{ CliFields, NotaryServerProperties };
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub struct Settings {
+    #[serde(flatten)]
+    pub config: NotaryServerProperties,
+}
+
+impl Settings {
+    pub fn new(cli_fields: &CliFields) -> Result<Self, ConfigError> {
+        let config_path = Path::new(&cli_fields.config_file);
+
+        let mut builder = Config::builder()
+            // Load base configuration
+            .add_source(File::from(config_path))
+            // Add in settings from environment variables (with a prefix of NOTARY_SERVER and '__' as separator)
+            .add_source(Environment::with_prefix("NOTARY_SERVER").prefix_separator("_").separator("__"));

```

Comment:
`prefix_separator` should be "__" and `separator` should be "_"; so that we have `NOTARY_SERVER__TLS_ENABLED`